mat/eigen.go

```diff
@@ -305,46 +307,54 @@ func (e *Eigen) complexEigenTo(dst *CDense, d *Dense) {
 	}
 }
 
-// VectorsTo returns the right eigenvectors of the decomposition. VectorsTo
-// will panic if the right eigenvectors were not computed during the factorization,
-// or if the factorization was not successful.
-//
-// The computed eigenvectors are normalized to have Euclidean norm equal to 1
+// VectorsTo returns the right eigenvectors of the decomposition, stored in-place
+// into dst. The computed eigenvectors are normalized to have Euclidean norm equal to 1
 // and largest component real.
-func (e *Eigen) VectorsTo(dst *CDense) *CDense {
+//
+// If dst is empty, VectorsTo will resize dst to be n×n. If dst is
+// non-empty, VectorsTo will panic if dst is not n×n. VectorsTo will also
+// panic if the eigenvectors were not computed during the factorization,
+// or if the receiver does not contain a successful factorization.
+func (e *Eigen) VectorsTo(dst *CDense) {
 	if !e.succFact() {
 		panic(badFact)
 	}
 	if e.kind&EigenRight == 0 {
 		panic(badNoVect)
 	}
-	if dst == nil {
-		dst = NewCDense(e.n, e.n, nil)
+	if dst.IsEmpty() {
+		dst.ReuseAs(e.n, e.n)
 	} else {
-		dst.reuseAsNonZeroed(e.n, e.n)
+		r, c := dst.Dims()
+		if r != e.n || c != e.n {
+			panic(ErrShape)
+		}
 	}
 	dst.Copy(e.rVectors)
-	return dst
 }
 
-// LeftVectorsTo returns the left eigenvectors of the decomposition. LeftVectorsTo
-// will panic if the left eigenvectors were not computed during the factorization,
-// or if the factorization was not successful.
-//
-// The computed eigenvectors are normalized to have Euclidean norm equal to 1
+// LeftVectorsTo returns the left eigenvectors of the decomposition, stored in-place
+// into dst. The computed eigenvectors are normalized to have Euclidean norm equal to 1
 // and largest component real.
-func (e *Eigen) LeftVectorsTo(dst *CDense) *CDense {
+//
+// If dst is empty, LeftVectorsTo will resize dst to be n×n. If dst is
+// non-empty, LeftVectorsTo will panic if dst is not n×n. LeftVectorsTo will also
```

Comment:
"When dst is non-empty, LeftVectorsTo will panic if dst is not n×n."

Reply:
Done.